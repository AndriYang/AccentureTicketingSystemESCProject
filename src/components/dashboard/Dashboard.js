import React, { Component } from 'react'
import { connect } from 'react-redux'
import { firestoreConnect } from 'react-redux-firebase'
import { compose } from 'redux'
import { Redirect } from 'react-router-dom'
import _ from "lodash";
import NewToDoListItem from "../ToDo/NewToDoListItem";

class Dashboard extends Component {
  state = {
    addFormVisible: false,
    addFormValue: "",
    checkedPassword: true,
    checkedRecruit: true,
    checkedTech: true
  };

  handleCheckTech = event =>{
    this.setState({checkedTech: !this.state.checkedTech});
  }

  renderCheckboxTech = () =>{
    return (<div>
      <form action="#">
      <label>
        <input
          name="isGoing"
          type="checkbox"
          checked={this.state.checkedTech}
          onChange={this.handleCheckTech} />
        <span>IT</span>
      </label>
      <br />
    </form>
    </div>
  );
  }

  handleCheckPassword = event =>{
    this.setState({checkedPassword: !this.state.checkedPassword});
  }

  renderCheckboxPassword = () =>{
    return (<div>
      <form action="#">
      <label>
        <input
          name="isGoing"
          type="checkbox"
          checked={this.state.checkedPassword}
          onChange={this.handleCheckPassword} />
        <span>Finance</span>
      </label>
      <br />
    </form>
    </div>
  );
  }

  handleCheckRecruit = event =>{
    this.setState({checkedRecruit: !this.state.checkedRecruit});
  }
  renderCheckboxRecruit = () =>{
    return (<div>
      <form action="#">
      <label>
        <input
          name="isGoing"
          type="checkbox"
          checked={this.state.checkedRecruit}
          onChange={this.handleCheckRecruit} />
        <span>General</span>
      </label>
      <br />
    </form>
    </div>
  );
  }

  renderToDos() {
    const { projects } = this.props;
    const toDos = _.map(projects, (value, key) => {
      return(
      <div>
        <NewToDoListItem key={key} todoId={key} todo={value}
          checkedTech={this.state.checkedTech}
          checkedRecruit={this.state.checkedRecruit}
          checkedPassword={this.state.checkedPassword}/>
      </div>
    );});


    if (!_.isEmpty(toDos)) {
      return toDos;
    }
    return (
      <div className="col s10 offset-s1 center-align">
        <img
          alt="Nothing was found"
          id="nothing-was-found"
          src="/img/nothing.png"
        />
        <h4>You have completed all the tasks</h4>
        <p>Start by clicking add button in the bottom of the screen</p>
      </div>
    );
  }





  render() {
    //console.log(this.props);
    const { projects, auth } = this.props;
    if (!auth.uid) return<Redirect to='/' />

    return (
      <div className="dashboard container">
        <div className="column">
          <div className="col s12 m6">
            <p>Showing categories:</p>
            {this.renderCheckboxTech()}
            {this.renderCheckboxPassword()}
            {this.renderCheckboxRecruit()}
            {this.renderToDos()}
          </div>
        </div>





      </div>
    )
  }
}

const mapStateToProps = (state) => {
  //console.log(state);
  return {
    projects: state.firestore.ordered.projects,
    auth: state.firebase.auth
  }
}

export default compose(
  connect(mapStateToProps),
  firestoreConnect([
    { collection: 'projects' }
  ])
)(Dashboard)
