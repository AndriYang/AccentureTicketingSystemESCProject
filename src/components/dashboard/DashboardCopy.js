import React, { Component } from 'react'
import Notifcations from './Notifications'
import { connect } from 'react-redux'
import { firestoreConnect } from 'react-redux-firebase'
import { compose } from 'redux'
import { Redirect } from 'react-router-dom'
import _ from "lodash";
import NewToDoListItem from "../ToDo/NewToDoListItem";

import styled from "styled-components";
import {SideNav, Nav, NavIcon } from "react-sidenav";
import { Icon } from "react-icons-kit";
import { ic_home as home } from "react-icons-kit/md/ic_home";
import { ic_reorder as simple } from "react-icons-kit/md/ic_reorder";
import { ic_donut_large as render } from "react-icons-kit/md/ic_donut_large";
import { MemoryRouter as Router, Route, Switch } from "react-router-dom";
import { AppContainer, Navigation, Body, Title } from "./containers";


const Text = styled.div`
  padding-left: 8px;
`;

const theme = {
  hoverBgColor: "#f5f5f5",
  selectionBgColor: "#f5f5f5",
  selectionIconColor: "#03A9F4"
};

export class Dashboard extends Component {
  state = {
    addFormVisible: false,
    addFormValue: "",
    checkedPassword: true,
    checkedRecruit: true,
    checkedTech: true,
    checkedRecruitment: true,
    checkedOther: true,
    selectedPath:'',
    checkedProcessing:true,
    checkedSolved:true
  };

  onItemSelection = (arg) => {
        this.setState({ selectedPath: arg.path })
    }
  handleCheckTech = event =>{
    console.log("checked it")
    this.setState({checkedTech: !this.state.checkedTech});
  }
  handleCheckPassword = event =>{
    console.log("checked finance")
    this.setState({checkedPassword: !this.state.checkedPassword});
  }
  handleCheckRecruit = event =>{
    console.log("checked general")
    this.setState({checkedRecruit: !this.state.checkedRecruit});
  }
  handleCheckRecruitment = event =>{
    console.log("checked recruitment")
    this.setState({checkedRecruitment: !this.state.checkedRecruitment});
  }
  handleCheckOther = event =>{
    console.log("checked other")
    this.setState({checkedOther: !this.state.checkedOther});
  }
  handleCheckProcessing = event =>{
    this.setState({checkedProcessing: !this.state.checkedProcessing});
  }
  handleCheckSolved = event =>{
    this.setState({checkedSolved: !this.state.checkedSolved});
  }
  renderToDos() {
    const { projects } = this.props;
    const toDos = _.map(projects, (value, key) => {
      return(
      <div>
        <NewToDoListItem key={key} todoId={key} todo={value}
          checkedTech={this.state.checkedTech}
          checkedRecruit={this.state.checkedRecruit}
          checkedRecruitment={this.state.checkedRecruitment}
          checkedPassword={this.state.checkedPassword}
          checkedOther={this.state.checkedOther}
          checkedProcessing={this.state.checkedProcessing}
          checkedSolved={this.state.checkedSolved}
          criteria={this.state.selectedPath}/>
      </div>
    );});


    if (!_.isEmpty(toDos)) {
      return toDos;
    }
    return (
      <div className="col s10 offset-s1 center-align">
        <img
          alt="Nothing was found"
          id="nothing-was-found"
          src="/img/nothing.png"
        />
        <h4>You have completed all the tasks</h4>
        <p>Start by clicking add button in the bottom of the screen</p>
      </div>
    );
  }

  renderRealtimeChat(){
    console.log("in realtimechat")
    if(this.state.selectedPath=="realTimeChat"){
      return(

        <Redirect to='/realtimechatagent' />

      )
    }else{
      return null
    }
  }

  renderNotifications(){
    if(this.state.selectedPath=="notifications"){
      const { projects, auth, notifications } = this.props;
      return(
        <div>
        <Notifcations notifications={notifications}/>
        </div>
      )
    }else{
      return null
    }
  }

  renderCheckBoxesCat(){
    if(this.state.selectedPath=="filtercategory"){
      return (
        <div className="row s8 m2" >
        <div onClick={this.handleCheckTech}>
        <button className={this.state.checkedTech ? "btn purple lighten-1 text-white":"btn purple lighten-3 text-white"} id="open-chat-bot" >
        IT
        </button></div>
        <div onClick={this.handleCheckPassword}>
        <button className={this.state.checkedPassword ? "btn purple lighten-1 text-white":"btn purple lighten-3 text-white"} id="open-chat-bot" >
        Finance
        </button></div>
        <div onClick={this.handleCheckRecruit}>
        <button className={this.state.checkedRecruit ? "btn purple lighten-1 text-white":"btn purple lighten-3 text-white"} id="open-chat-bot" >
        General
        </button></div>
        <div onClick={this.handleCheckRecruitment}>
        <button className={this.state.checkedRecruitment ? "btn purple lighten-1 text-white":"btn purple lighten-3 text-white"} id="open-chat-bot" >
        Recruitment
        </button></div>
        <div onClick={this.handleCheckOther}>
        <button className={this.state.checkedOther ? "btn purple lighten-1 text-white":"btn purple lighten-3 text-white"} id="open-chat-bot" >
        Other
        </button></div>
        </div>
      )
    }
  }

  renderCheckBoxesProg(){
    if(this.state.selectedPath=="filterprogress"){
      return (
        <div className="row s8 m2" >
        <div id="status" onClick={this.handleCheckProcessing}>
        <button className={this.state.checkedProcessing ? "btn purple lighten-1 text-white":"btn purple lighten-3 text-white"} >
        Processing
        </button></div>
        <div id="status" onClick={this.handleCheckSolved}>
        <button className={this.state.checkedSolved ? "btn purple lighten-1 text-white":"btn purple lighten-3 text-white"} >
        Solved
        </button></div>
        </div>
      )
    }
  }

  render() {
    //console.log(this.props);
    const { projects, auth, notifications } = this.props;
    if (!auth.uid) return<Redirect to='/' />

    return (
      <AppContainer>
      <Navigation>
      <SideNav theme={theme} defaultSelectedPath={"home"}
                onItemSelection={this.onItemSelection}>
        <Nav id="home">
          <NavIcon>
            <Icon icon={home} />
          </NavIcon>
          <Text>Home</Text>
        </Nav>
        <Nav id="basic">
          <NavIcon>
            <Icon icon={simple} />
          </NavIcon>
          <Text>All tickets</Text>
        </Nav>
        <Nav id="realTimeChat">
          <NavIcon>
            <Icon icon={render} />
          </NavIcon>
          <Text>RealTime Chat</Text>
        </Nav>
        <Nav id="notifications">
          <NavIcon>
            <Icon icon={render} />
          </NavIcon>
          <Text>Notifications</Text>
        </Nav>
        <Nav id="filtercategory">
          <NavIcon>
            <Icon icon={render} />
          </NavIcon>
          <Text>Filter by Category</Text>
        </Nav>
        <Nav id="filterprogress">
          <NavIcon>
            <Icon icon={render} />
          </NavIcon>
          <Text>Filter by Progress</Text>
        </Nav>
      </SideNav>
      </Navigation>
      <Body class="col">
          <div className="dashboard container">
            <div class="row" >
            {this.renderRealtimeChat()}
            {this.renderCheckBoxesProg()}
            {this.renderCheckBoxesCat()}
            {this.renderNotifications()}
              </div>
              <div class="divider"></div>
              <div><span class="flow-text">Tickets:</span></div>
              <div class="divider"></div>
              <div class="card purple lighten-5">
                <div className="col s12 m6" >
                  {this.renderToDos()}
                </div>
              </div>
          </div>
        </Body>
      </AppContainer>

    )
  }
}

const mapStateToProps = (state) => {
  //console.log(state);
  return {
    projects: state.firestore.ordered.projects,
    auth: state.firebase.auth,
    notifications: state.firestore.ordered.notifications
  }
}

export default compose(
  connect(mapStateToProps),
  firestoreConnect([
    { collection: 'projects' },
    {collection: 'notifications', limit:5, orderBy:['time', 'desc']}
  ])
)(Dashboard)
