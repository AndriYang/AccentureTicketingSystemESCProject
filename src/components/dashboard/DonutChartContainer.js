import React from 'react';
import 'hammerjs';

import {
  Chart,
  ChartLegend,
  ChartSeries,
  ChartSeriesItem,
  ChartSeriesLabels
} from '@progress/kendo-react-charts';

import { donutChartData } from './data/salesData';

/* This function's outputted string determines the label contents */
const labelTemplate = (e) => (e.category + '\n'  + (e.percentage*100) +'%');
export const DonutChartContainer = () => (
  <Chart style={{ height: 300 }}>
    <ChartSeries>
      <ChartSeriesItem type="donut" data={donutChartData} categoryField="foodType" field="percentSold" padding={0}>
        <ChartSeriesLabels color="#fff" background="none" content={labelTemplate} />
      </ChartSeriesItem>
    </ChartSeries>
    <ChartLegend visible={false} />
  </Chart>

);
