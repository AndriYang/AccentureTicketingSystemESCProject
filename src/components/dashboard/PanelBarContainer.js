import React from 'react';
import { PanelBar, PanelBarItem } from '@progress/kendo-react-layout';
import { panelBarData } from './data/salesData';

const imageUrl = (imageName) => ('img/' + imageName + '.jpg');

export const PanelBarContainer = () => (
  <PanelBar >
    <PanelBarItem expanded={true} title="My Teammates">
      <div>
        {panelBarData.teammates.map((item, idx) => (
          <div className={idx === 0 ? 'teammate k-state-selected' : 'teammate'} id={item.firstName + ' ' + item.lastName} key={idx}>
            <img src={imageUrl(item.firstName)} alt={item.firstName + ' ' + item.lastName} />
            <span className="mate-info">
              <h2>{item.firstName + ' ' + item.lastName}</h2>
              <p>{item.position}</p>
            </span>
          </div>
        ))}
      </div>
    </PanelBarItem>
    <PanelBarItem title={'Projects'} >
      <PanelBarItem title={'Sales Reports'} >
        {panelBarData.salesReports.map((item, idx) => (
          <PanelBarItem title={item.title} key={idx} />
        ))}
      </PanelBarItem >
    </PanelBarItem>
    <PanelBarItem title="Communication" disabled={true} />
  </PanelBar>
)
