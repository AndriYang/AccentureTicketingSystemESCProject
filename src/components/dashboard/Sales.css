h1 {
  color:rgb(63, 81, 181);
  font-weight: bold;
}
.buttons-right>button {
  float: right;
  margin: 5px;
}
.app-container {
  padding: 25px;
}

.percentage-number {
  color: rgb(63, 81, 181);
  font-size: 50px;
  font-weight: bold;
}
.percentage-sign {
  color: rgb(63, 81, 181);
  font-size: 40px;
}
.percentage-container {
  text-align: center;
}
.percentage-container:first-child {
  padding-bottom: 3em;
}

.teammate:after {
  content: ".";
  display: block;
  height: 0;
  line-height: 0;
  clear: both;
  visibility: hidden;
}
.teammate h2 {
  font-size: 1.3em;
  font-weight: normal;
  padding-top: 17px;
  margin: 0;
}
.teammate p {
  margin: 0;
  font-size: .7em;
}
.teammate img {
  display: inline-block;
  vertical-align: top;
  width: 50px;
  height: 50px;
  margin: 10px;
  border: 1px solid #ccc;
  border-radius: 50%;
}
.mate-info {
  display: inline-block;
  vertical-align: top;
}
.panelbar-wrapper {
  max-width: 300px;
  margin: 0 auto;
}
