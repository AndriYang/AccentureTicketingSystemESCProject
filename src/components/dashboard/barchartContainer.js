import React from 'react';
import 'hammerjs';

import {
  Chart,
  ChartSeries,
  ChartSeriesItem,
  ChartCategoryAxis,
  ChartCategoryAxisItem,
  ChartLegend,
  ChartValueAxis,
  ChartValueAxisItem
} from '@progress/kendo-react-charts';

import { barChartQ4Months, barChartMonthlyPercentages } from './data/salesData';

export const BarChartContainer = () => (
  <Chart style={{ height: 288 }}>
    <ChartLegend visible={false} />
    <ChartCategoryAxis>
      <ChartCategoryAxisItem categories={barChartQ4Months} startAngle={45} />
    </ChartCategoryAxis>
    <ChartSeries>
      {
        barChartMonthlyPercentages.map((item, idx) => (
          <ChartSeriesItem key={idx} type="column" data={item.data} name={item.name} gap={2}/>
        ))}
    </ChartSeries>
    <ChartValueAxis skip={4}>
      <ChartValueAxisItem color="#888" skip={2} />
    </ChartValueAxis>
  </Chart>
);
