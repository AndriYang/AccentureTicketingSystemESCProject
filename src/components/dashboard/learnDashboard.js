
import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { Dialog, DialogActionsBar } from '@progress/kendo-react-dialogs';
import { Input } from '@progress/kendo-react-inputs';
import { Button } from '@progress/kendo-react-buttons';
import { Ripple } from '@progress/kendo-react-ripple';
import { savePDF } from '@progress/kendo-react-pdf';

import {DonutChartContainer} from './DonutChartContainer';
import { BarChartContainer } from './barchartContainer';
import { GridContainer } from './GridContainer';
import { PanelBarContainer } from './PanelBarContainer';

import '@progress/kendo-theme-material/dist/all.css';
import 'bootstrap-4-grid/css/grid.min.css';

import { connect } from 'react-redux'
import { firestoreConnect } from 'react-redux-firebase'
import { compose } from 'redux'
import _ from "lodash";

import styled from "styled-components";
import { MemoryRouter as Router, Route, Switch } from "react-router-dom";
import { AppContainer, Navigation, Body, Title } from "./containers";

import SalesData from './data/salesDataClass'

class App extends Component {
  constructor(props) {
    super(props);
    this.appContainer = React.createRef();
    this.state = {
      showDialog: false,
      total_submission: 245,
      resolve_speed: 2,
      first_response: 1,
      urgent_tickets: 5,
      percentage: 94
    }
  }
  handlePDFExport = () => {
    savePDF(ReactDOM.findDOMNode(this.appContainer), { paperSize: 'auto' });
  }
  handleShare = () => {
    this.setState({
      showDialog: !this.state.showDialog
    }, function(){

    })
  }
  renderDonutChartContainer=()=>{
    return <DonutChartContainer />
  }
  renderBarChartContainer=()=>{
    return <BarChartContainer />
  }
  renderHeader=()=>{
    const date = new Date();
    const year = date.getYear();
    var Q = 1;
    if (date.getMonth()<=5&&date.getMonth()>2){
      Q = 2;
    }
    if (date.getMonth()<=8&&date.getMonth()>5){
      Q = 3;
    }
    if (date.getMonth()<=11&&date.getMonth()>8){
      Q = 4;
    }
    return (
      <div className="col-xs-6 col-sm-6 col-md-6 col-lg-6 col-xl-6">
        <h1>Ticket Dashboard | Q{Q} {year+1900}</h1>
      </div>
    )
  }
  render() {
    return (
        <div className="bootstrap-wrapper">
          <div className="app-container container" ref={(el) => this.appContainer = el}>
            <div className="row">
              {this.renderHeader()}
              <div className="col-xs-6 col-sm-6 col-md-6 col-lg-6 col-xl-6 buttons-right">

                <Button primary={true} onClick={this.handlePDFExport}>Export as PDF</Button>
              </div>
            </div>
            <div className="row">
              <div className="col-xs-9 col-sm-9 col-md-9 col-lg-12 col-xl-12">
                <div className="row">
                  <div className="col-xs-6 col-sm-6 col-md-4 col-lg-4 col-xl-4">
                  <SalesData projects={this.props.projects} type="donutChart"/>
                  </div>
                  <div className="col-xs-12 col-sm-12 col-md-6 col-lg-6 col-xl-6">
                  <SalesData projects={this.props.projects} type="barChart"/>
                  </div>
                  <div className="col-xs-12 col-sm-12 col-md-6 col-lg-2 col-xl-2">
                  <SalesData projects={this.props.projects} type="totalSubmission"/>
                  </div>
                </div>
                <div className="row">
                    <div className="col-xs-6 col-sm-6 col-md-2 col-lg-3 col-xl-3">
                    <SalesData projects={this.props.projects} type="ticketSolved"/>
                  </div>
                  <div className="col-xs-6 col-sm-6 col-md-2 col-lg-3 col-xl-3">
                  <SalesData projects = {this.props.projects} type="solveSpeed"/>
                  </div>

                  <div className="col-xs-6 col-sm-6 col-md-2 col-lg-3 col-xl-3">
                  <SalesData projects={this.props.projects} type="approachingDeadline"/>
                  </div>
                </div>
              </div>
            </div>
            {this.state.showDialog &&
              <Dialog title={"Share this report"} onClose={this.handleShare}>
                <p>Please enter the email address/es of the recipient/s.</p>
                <Input placeholder="example@progress.com" />
                <DialogActionsBar>
                  <Button primary={true} onClick={this.handleShare}>Share</Button>
                  <Button onClick={this.handleShare}>Cancel</Button>
                </DialogActionsBar>
              </Dialog>
            }
          </div>
        </div>
    );
  }
}

const mapStateToProps = (state) => {
  console.log(state);
  return {
    projects: state.firestore.ordered.projects,
    auth: state.firebase.auth
  }
}
export default compose(
  connect(mapStateToProps),
  firestoreConnect([
    { collection: 'projects' }
  ])
)(App);
