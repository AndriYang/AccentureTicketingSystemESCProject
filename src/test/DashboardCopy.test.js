import {Dashboard} from '../components/dashboard/DashboardCopy'
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { Link } from 'react-router-dom';
import { create } from "react-test-renderer";

configure({ adapter: new Adapter() });
it('print the initial state',() => {
  const data = new Dashboard();
  expect(data).toMatchSnapshot();
})





// const example = jest.genMockFromModule('../components/dashboard/Dashboard');
// test('should run example code', () => {
//   // creates a new mocked function with no formal arguments.
//   expect(example.function.name).toEqual('renderCheckboxPassword');
//   expect(example.function.length).toEqual(0);})

describe("Dashboard component", () => {
  test("it matches the snapshot", () => {
    const component = create(<Dashboard />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
