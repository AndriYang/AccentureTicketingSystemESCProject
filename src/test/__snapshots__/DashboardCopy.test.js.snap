// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component it matches the snapshot 1`] = `
<div
  className="sc-bZQynM fhhqqT"
>
  <div
    className="sc-htoDjs fHajFs"
  >
    <div
      className="sc-EHOje chXdSi"
    >
      <div
        onClick={[Function]}
      >
        <div
          className="sc-bdVaJa fMlwjV"
          selected={true}
          style={Object {}}
        >
           
          <div
            className="sc-htpNat crmRjJ"
            selected={true}
            style={Object {}}
          >
            <i
              fill="currentColor"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <svg
                fill="currentColor"
                height={16}
                style={
                  Object {
                    "display": "inline-block",
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 24 24"
                width={16}
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
            </i>
          </div>
          <div
            className="sc-jTzLTM gGGooV"
          >
            Home
          </div>
        </div>
      </div>
      <div
        onClick={[Function]}
      >
        <div
          className="sc-bdVaJa jUQUWx"
          selected={false}
          style={Object {}}
        >
           
          <div
            className="sc-htpNat kVxbbl"
            selected={false}
            style={Object {}}
          >
            <i
              fill="currentColor"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <svg
                fill="currentColor"
                height={16}
                style={
                  Object {
                    "display": "inline-block",
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 24 24"
                width={16}
              >
                <path
                  d="M3 15h18v-2H3v2zm0 4h18v-2H3v2zm0-8h18V9H3v2zm0-6v2h18V5H3z"
                />
              </svg>
            </i>
          </div>
          <div
            className="sc-jTzLTM gGGooV"
          >
            All tickets
          </div>
        </div>
      </div>
      <div
        onClick={[Function]}
      >
        <div
          className="sc-bdVaJa jUQUWx"
          selected={false}
          style={Object {}}
        >
           
          <div
            className="sc-htpNat kVxbbl"
            selected={false}
            style={Object {}}
          >
            <i
              fill="currentColor"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <svg
                fill="currentColor"
                height={16}
                style={
                  Object {
                    "display": "inline-block",
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 24 24"
                width={16}
              >
                <path
                  d="M11 5.08V2c-5 .5-9 4.81-9 10s4 9.5 9 10v-3.08c-3-.48-6-3.4-6-6.92s3-6.44 6-6.92zM18.97 11H22c-.47-5-4-8.53-9-9v3.08C16 5.51 18.54 8 18.97 11zM13 18.92V22c5-.47 8.53-4 9-9h-3.03c-.43 3-2.97 5.49-5.97 5.92z"
                />
              </svg>
            </i>
          </div>
          <div
            className="sc-jTzLTM gGGooV"
          >
            RealTime Chat
          </div>
        </div>
      </div>
      <div
        onClick={[Function]}
      >
        <div
          className="sc-bdVaJa jUQUWx"
          selected={false}
          style={Object {}}
        >
           
          <div
            className="sc-htpNat kVxbbl"
            selected={false}
            style={Object {}}
          >
            <i
              fill="currentColor"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <svg
                fill="currentColor"
                height={16}
                style={
                  Object {
                    "display": "inline-block",
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 24 24"
                width={16}
              >
                <path
                  d="M11 5.08V2c-5 .5-9 4.81-9 10s4 9.5 9 10v-3.08c-3-.48-6-3.4-6-6.92s3-6.44 6-6.92zM18.97 11H22c-.47-5-4-8.53-9-9v3.08C16 5.51 18.54 8 18.97 11zM13 18.92V22c5-.47 8.53-4 9-9h-3.03c-.43 3-2.97 5.49-5.97 5.92z"
                />
              </svg>
            </i>
          </div>
          <div
            className="sc-jTzLTM gGGooV"
          >
            Notifications
          </div>
        </div>
      </div>
      <div
        onClick={[Function]}
      >
        <div
          className="sc-bdVaJa jUQUWx"
          selected={false}
          style={Object {}}
        >
           
          <div
            className="sc-htpNat kVxbbl"
            selected={false}
            style={Object {}}
          >
            <i
              fill="currentColor"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <svg
                fill="currentColor"
                height={16}
                style={
                  Object {
                    "display": "inline-block",
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 24 24"
                width={16}
              >
                <path
                  d="M11 5.08V2c-5 .5-9 4.81-9 10s4 9.5 9 10v-3.08c-3-.48-6-3.4-6-6.92s3-6.44 6-6.92zM18.97 11H22c-.47-5-4-8.53-9-9v3.08C16 5.51 18.54 8 18.97 11zM13 18.92V22c5-.47 8.53-4 9-9h-3.03c-.43 3-2.97 5.49-5.97 5.92z"
                />
              </svg>
            </i>
          </div>
          <div
            className="sc-jTzLTM gGGooV"
          >
            Filter by Category
          </div>
        </div>
      </div>
      <div
        onClick={[Function]}
      >
        <div
          className="sc-bdVaJa jUQUWx"
          selected={false}
          style={Object {}}
        >
           
          <div
            className="sc-htpNat kVxbbl"
            selected={false}
            style={Object {}}
          >
            <i
              fill="currentColor"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <svg
                fill="currentColor"
                height={16}
                style={
                  Object {
                    "display": "inline-block",
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 24 24"
                width={16}
              >
                <path
                  d="M11 5.08V2c-5 .5-9 4.81-9 10s4 9.5 9 10v-3.08c-3-.48-6-3.4-6-6.92s3-6.44 6-6.92zM18.97 11H22c-.47-5-4-8.53-9-9v3.08C16 5.51 18.54 8 18.97 11zM13 18.92V22c5-.47 8.53-4 9-9h-3.03c-.43 3-2.97 5.49-5.97 5.92z"
                />
              </svg>
            </i>
          </div>
          <div
            className="sc-jTzLTM gGGooV"
          >
            Filter by Progress
          </div>
        </div>
      </div>
      <div
        onClick={[Function]}
      >
        <div
          className="sc-bdVaJa jUQUWx"
          selected={false}
          style={Object {}}
        >
           
          <div
            className="sc-htpNat kVxbbl"
            selected={false}
            style={Object {}}
          >
            <i
              fill="currentColor"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <svg
                fill="currentColor"
                height={16}
                style={
                  Object {
                    "display": "inline-block",
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 24 24"
                width={16}
              >
                <path
                  d="M11 5.08V2c-5 .5-9 4.81-9 10s4 9.5 9 10v-3.08c-3-.48-6-3.4-6-6.92s3-6.44 6-6.92zM18.97 11H22c-.47-5-4-8.53-9-9v3.08C16 5.51 18.54 8 18.97 11zM13 18.92V22c5-.47 8.53-4 9-9h-3.03c-.43 3-2.97 5.49-5.97 5.92z"
                />
              </svg>
            </i>
          </div>
          <div
            className="sc-jTzLTM gGGooV"
          >
            Filter by Deadline
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="col"
    className="sc-dnqmqq fZOEMv"
  >
    <div
      className="dashboard container"
    >
      <div
        class="row"
      />
      <div
        class="divider"
      />
      <div>
        <span
          class="flow-text"
        >
          Tickets:
        </span>
      </div>
      <div
        class="divider"
      />
      <div
        class="card purple lighten-5"
      >
        <div
          className="col s12 m6"
        >
          <div
            className="col s10 offset-s1 center-align"
          >
            <img
              alt="Nothing was found"
              id="nothing-was-found"
              src="/img/nothing.png"
            />
            <h4>
              You have completed all the tasks
            </h4>
            <p>
              Start by clicking add button in the bottom of the screen
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`print the initial state 1`] = `
Dashboard {
  "context": undefined,
  "handleCheckDeadline": [Function],
  "handleCheckOther": [Function],
  "handleCheckPassword": [Function],
  "handleCheckProcessing": [Function],
  "handleCheckRecruit": [Function],
  "handleCheckRecruitment": [Function],
  "handleCheckSolved": [Function],
  "handleCheckTech": [Function],
  "onItemSelection": [Function],
  "props": undefined,
  "refs": Object {},
  "state": Object {
    "addFormValue": "",
    "addFormVisible": false,
    "checkedDeadline": true,
    "checkedOther": true,
    "checkedPassword": true,
    "checkedProcessing": true,
    "checkedRecruit": true,
    "checkedRecruitment": true,
    "checkedSolved": true,
    "checkedTech": true,
    "selectedPath": "",
  },
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;
